### PR TITLE
Add check for $ZSH creation permission

### PR DESCRIPTION
**Pull request by RichardLuo0**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.

## Changes:

- Check if you can really create $ZSH folder.

## Other comments:
The `if [ -d "$ZSH" ]` give false even if some folders in the middle are inaccessible.

### CODE REVIEW

File: tools/install.sh [hunk 1]
```diff
@@ -254,6 +255,7 @@ main() {
       --unattended) RUNZSH=no; CHSH=no ;;
       --skip-chsh) CHSH=no ;;
       --keep-zshrc) KEEP_ZSHRC=yes ;;
+      --skip-git-clone) SKIP_GIT_CLONE=yes ;;
```
Comment (mcornella) [inline tools/install.sh:258]: What does this do?

Comment (RichardLuo0) [reply tools/install.sh:258]: If for some reason user clone the repo manually before running the script, he can skip git clone process. For example, user wants to share a same clone among multiple linux users.

**Comment by larson-carter** [comment]
Circling back to this older PR, I do think that it is useful. However we would need more testing. I would also need more understanding as to the edge cases of this?

**Comment by RichardLuo0** [comment]
It seems git will fail to clone anyway if the $ZSH is not writeable. I think this is not needed anymore unless you would like an early error.
It has been several years, so I don't quite remember why I did this PR at that time.

The second commit about adding skip-git-clone is an accident push. It is not related to this PR. I will revert it if you want to merge.

**Pull request closed without merging**